### PR TITLE
[DSM] Remove expected_replica_count

### LINKED ISSUE #407: [DSM] Remove extend replicas

### PR DESCRIPTION
**Pull request by MyronFanQiu**
Fix #407

### CODE REVIEW

File: cstrml/market/src/lib.rs [hunk 1]
```diff
@@ -266,7 +264,7 @@ pub trait Config: system::Config {
     type FileDuration: Get<BlockNumber>;
 
     /// File base replica. Use 4 for now
-    type InitialReplica: Get<u32>;
+    type FileReplica: Get<u32>;
```
Comment (MyronFanQiu) [inline cstrml/market/src/lib.rs:267]: Here

File: cstrml/market/src/lib.rs [hunk 2]
```diff
@@ -701,7 +698,7 @@ impl<T: Config> Module<T> {
         Self::update_files_size(file_info.file_size, prev_reported_group_count, used_info.reported_group_count);
 
         let claim_block = curr_bn.min(file_info.expired_on);
-        let target_reward_count = file_info.replicas.len().min(file_info.expected_replica_count as usize) as u32;
+        let target_reward_count = file_info.replicas.len().min(T::FileReplica::get() as usize) as u32;
```
Comment (MyronFanQiu) [inline cstrml/market/src/lib.rs:701]: Here

File: cstrml/market/src/lib.rs [hunk 3]
```diff
@@ -928,18 +925,14 @@ impl<T: Config> Module<T> {
...
-                    // we keep the original amount and expected_replica_count
+                    // we keep the original amount
                     file_info.expired_on = 0;
                 } else {
                     file_info.expired_on = curr_bn + T::FileDuration::get();
                 }
                 file_info.claimed_at = *curr_bn;
             }
             file_info.amount += amount.clone();
-            if extend_replica {
```
Comment (MyronFanQiu) [inline cstrml/market/src/lib.rs:939]: Here

**Review by zikunfan — APPROVED** [review]
lgtm

**Pull request merged**